edb/schema/pointers.py

```diff
@@ -1628,6 +1628,30 @@ def as_inherited_ref_cmd(
         return cmd
 
 
+class DeletePointer(
+    referencing.DeleteReferencedInheritingObject[Pointer_T],
+    PointerCommand[Pointer_T],
+):
+    def _canonicalize(
+        self,
+        schema: s_schema.Schema,
+        context: sd.CommandContext,
+        scls: Pointer_T,
+    ) -> List[sd.Command]:
+        commands = super()._canonicalize(schema, context, scls)
+
+        # Any union type that references this field needs to have it
+        # deleted.
+        unions = schema.get_referrers(
+            self.scls, scls_type=Pointer, field_name='union_of')
+        for union in unions:
+            group, _, _ = union.init_delta_branch(
```

Comment:
I think you need to run `_canonicalize` on `self_cmd` returned by `init_delta_branch`.